README.md
# NAME

Dist::Zilla::Plugin::Stenciller::MojoliciousTests - Create Mojolicious tests from text files parsed with Stenciller

<div>
    <p>
    <img src="https://img.shields.io/badge/perl-5.10+-blue.svg" alt="Requires Perl 5.10+" />
    <a href="https://travis-ci.org/Csson/p5-Dist-Zilla-Plugin-Stenciller-MojoliciousTests"><img src="https://api.travis-ci.org/Csson/p5-Dist-Zilla-Plugin-Stenciller-MojoliciousTests.svg?branch=master" alt="Travis status" /></a>
    <a href="http://cpants.cpanauthors.org/dist/Dist-Zilla-Plugin-Stenciller-MojoliciousTests-0.0200"><img src="https://badgedepot.code301.com/badge/kwalitee/Dist-Zilla-Plugin-Stenciller-MojoliciousTests/0.0200" alt="Distribution kwalitee" /></a>
    <a href="http://matrix.cpantesters.org/?dist=Dist-Zilla-Plugin-Stenciller-MojoliciousTests%200.0200"><img src="https://badgedepot.code301.com/badge/cpantesters/Dist-Zilla-Plugin-Stenciller-MojoliciousTests/0.0200" alt="CPAN Testers result" /></a>
    <img src="https://img.shields.io/badge/coverage-98.1%-yellow.svg" alt="coverage 98.1%" />
    </p>
</div>

# VERSION

Version 0.0200, released 2016-03-23.

# SYNOPSIS

    ; in dist.ini
    ; these are the defaults:
    [Stenciller::MojoliciousTests]
    source_directory = examples/source
    file_pattern = .+\.stencil
    template_file = examples/source/template.test
    output_directory = t

# DESCRIPTION

Dist::Zilla::Plugin::Stenciller::MojoliciousTests uses [Stenciller](https://metacpan.org/pod/Stenciller) and [Stenciller::Plugin::ToMojoliciousTest](https://metacpan.org/pod/Stenciller::Plugin::ToMojoliciousTest) to turn
stencil files in `source_directory` (that matches the `file_pattern`) into
test files in `output_directory` by applying the `template_file`.

This [Dist::Zilla](https://metacpan.org/pod/Dist::Zilla) plugin does the `FileGatherer` role.

# ATTRIBUTES

## source\_directory

Path to where the stencil files are.

## output\_directory

Path to where the generated files are saved.

## file\_pattern

This is put inside a regular expression (with start and end anchors) to find stencil files in the `source_directory`. The output files
will have the same basename, but the suffix is replaced by `t`.

## template\_file

The template file should contain use statements and such. The transformed contents returned from [Stenciller::Plugin::ToMojoliciousTest](https://metacpan.org/pod/Stenciller::Plugin::ToMojoliciousTest) will be placed after
the contents of `template_file`. The template file is applied to each stencil file, so the number of generated test files is equal
to the number of stencil files.

# SEE ALSO

- [Stenciller](https://metacpan.org/pod/Stenciller)
- [Stenciller::Plugin::ToMojoliciousTest](https://metacpan.org/pod/Stenciller::Plugin::ToMojoliciousTest)

# SOURCE

[https://github.com/Csson/p5-Dist-Zilla-Plugin-Stenciller-MojoliciousTests](https://github.com/Csson/p5-Dist-Zilla-Plugin-Stenciller-MojoliciousTests)

# HOMEPAGE

[https://metacpan.org/release/Dist-Zilla-Plugin-Stenciller-MojoliciousTests](https://metacpan.org/release/Dist-Zilla-Plugin-Stenciller-MojoliciousTests)

# AUTHOR

Erik Carlsson <info@code301.com>

# COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Erik Carlsson.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for Dist-Zilla-Plugin-Stenciller-MojoliciousTests

{{$NEXT}}
 [API Changes]

 [Bug Fixes]

 [Documentation]

 [Enhancements]

 [New Features]

 [Other Changes]

 [Requirements]

 [Stats & Dependency Changes]

 [Testing]

0.0200 2016-03-23T22:10:20Z UTC
 [Bug Fixes]
 - Fix compatibility with latest Stenciller (0.1400) (fixes #1).

 [Requirements]
 - Added missing prereq: Dist::Zilla::Role::FileGatherer
 - Perl requirement lowered to 5.10 (from 5.14).

 [Stats & Dependency Changes]
 - 2 files changed, 83 insertions(+), 76 deletions(-)
 - (run req) + Dist::Zilla::Role::FileGatherer (any)
 - (run req) + Types::Path::Tiny 0.005
 - (run req) + Types::Standard 1.000005
 - (run req) ~ Moose 2.1400 --> 2.1405
 - (run req) ~ Path::Tiny (any) --> 0.072
 - (run req) ~ Stenciller 0.1300 --> 0.1400
 - (run req) ~ Stenciller::Plugin::ToMojoliciousTest (any) --> 0.0102
 - (run req) ~ perl 5.014000 --> 5.010000
 - (dev req) + Badge::Depot 0.0103
 - (dev req) + Badge::Depot::Plugin::Coverage (any)
 - (dev req) + Badge::Depot::Plugin::Cpantesters (any)
 - (dev req) + Badge::Depot::Plugin::Kwalitee (any)
 - (dev req) + Badge::Depot::Plugin::Perl (any)
 - (dev req) + Badge::Depot::Plugin::Travis (any)
 - (dev req) + Dist::Zilla 5.000
 - (dev req) + Dist::Zilla::Plugin::Authority 1.009
 - (dev req) + Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.007
 - (dev req) + Dist::Zilla::Plugin::ChangeStats::Dependencies::Git 0.0100
 - (dev req) + Dist::Zilla::Plugin::Git 2.036
 - (dev req) + Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch 0.013
 - (dev req) + Dist::Zilla::Plugin::Git::Commit (any)
 - (dev req) + Dist::Zilla::Plugin::NextRelease::Grouped 0.0200
 - (dev req) + Dist::Zilla::Plugin::RewriteVersion::Transitional 0.007
 - (dev req) + Dist::Zilla::Plugin::Test::EOF 0.0501
 - (dev req) + Dist::Zilla::Plugin::Test::EOL 0.18
 - (dev req) + Dist::Zilla::Plugin::Test::Kwalitee::Extra v0.2.1
 - (dev req) + Dist::Zilla::Plugin::Test::NoTabs 0.15
 - (dev req) + Dist::Zilla::Plugin::Test::Version 1.05
 - (dev req) + Pod::Weaver::Section::GenerateSection 1.01
 - (dev req) + Test::EOF (any)
 - (dev req) + Test::EOL (any)
 - (dev req) + Test::More 0.88
 - (dev req) + Test::NoTabs 1.4
 - (dev req) + Test::Version 1
 - (dev req) ~ Dist::Zilla::Plugin::CheckChangesHasContent (any) --> 0.008
 - (dev req) ~ Dist::Zilla::Plugin::Clean (any) --> 0.07
 - (dev req) ~ Dist::Zilla::Plugin::CopyFilesFromBuild (any) --> 0.151680
 - (dev req) ~ Dist::Zilla::Plugin::Git::Contributors (any) --> 0.017
 - (dev req) ~ Dist::Zilla::Plugin::GithubMeta (any) --> 0.54
 - (dev req) ~ Dist::Zilla::Plugin::InstallRelease (any) --> 0.008
 - (dev req) ~ Dist::Zilla::Plugin::MetaProvides::Package (any) -->
   2.003001
 - (dev req) ~ Dist::Zilla::Plugin::PodWeaver (any) --> 4.006
 - (dev req) ~ Dist::Zilla::Plugin::ReadmeAnyFromPod (any) --> 0.150250
 - (dev req) ~ Dist::Zilla::Plugin::RunExtraTests (any) --> 0.028
 - (dev req) ~ Dist::Zilla::Plugin::Test::Compile (any) --> 2.054
 - (dev req) ~ Dist::Zilla::Plugin::Test::ReportPrereqs (any) --> 0.021
 - (dev req) ~ Dist::Zilla::Plugin::TravisYML (any) --> 1.15
 - (dev req) ~ Pod::Weaver::Section::Badges (any) --> 0.0401
 - (dev req) ~ Test::Warnings (any) --> 0.026
 - (dev req) - Dist::Iller::Config::Author::CSSON
 - (dev req) - Dist::Zilla::Plugin::MetaProvides::Class
 - (dev req) - Dist::Zilla::Plugin::NextRelease
 - (dev req) - Dist::Zilla::Plugin::NextVersion::Semantic
 - (dev req) - Dist::Zilla::Plugin::OurPkgVersion
 - (dev req) - Dist::Zilla::Plugin::PodnameFromClassname
 - (dev req) - Dist::Zilla::Plugin::Prereqs::Plugins
 - (dev req) - Dist::Zilla::Plugin::PreviousVersion::Changelog
 - (dev req) - Dist::Zilla::Plugin::ReversionOnRelease
 - (dev req) - Pod::Weaver::Section::Homepage::DefaultCPAN
 - (dev req) - Pod::Weaver::Section::Source::DefaultGitHub
 - (dev sug) + Dist::Iller 0.1408
 - (dev sug) + Dist::Iller::Config::Author::CSSON 0.0309

0.0101 2016-01-25T21:42:53Z UTC
 [Other Changes]
 - Rebuilt with current Dist::Iller::Config::Author::CSSON

 [Requirements]
 - Added missing prereq: Moose
 - New prereq: namespace::autoclean

0.0100 2015-11-23T09:48:00Z UTC
 - Initial release


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.043.
use strict;
use warnings;

use 5.010000;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Create Mojolicious tests from text files parsed with Stenciller",
  "AUTHOR" => "Erik Carlsson <info\@code301.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Stenciller-MojoliciousTests",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010000",
  "NAME" => "Dist::Zilla::Plugin::Stenciller::MojoliciousTests",
  "PREREQ_PM" => {
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Moose" => "2.1405",
    "Path::Tiny" => "0.072",
    "Stenciller" => "0.1400",
    "Stenciller::Plugin::ToMojoliciousTest" => "0.0102",
    "Types::Path::Tiny" => "0.005",
    "Types::Standard" => "1.000005",
    "namespace::autoclean" => "0.22"
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Mojolicious" => "5.00",
    "String::Cushion" => 0,
    "Syntax::Feature::Qi" => 0,
    "Test::DZil" => 0,
    "Test::Differences" => 0,
    "Test::Mojo::Trim" => 0,
    "Test::More" => "0.96",
    "syntax" => 0
  },
  "VERSION" => "0.0201",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::Role::FileGatherer" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Mojolicious" => "5.00",
  "Moose" => "2.1405",
  "Path::Tiny" => "0.072",
  "Stenciller" => "0.1400",
  "Stenciller::Plugin::ToMojoliciousTest" => "0.0102",
  "String::Cushion" => 0,
  "Syntax::Feature::Qi" => 0,
  "Test::DZil" => 0,
  "Test::Differences" => 0,
  "Test::Mojo::Trim" => 0,
  "Test::More" => "0.96",
  "Types::Path::Tiny" => "0.005",
  "Types::Standard" => "1.000005",
  "namespace::autoclean" => "0.22",
  "syntax" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
